resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/PerformanceCriticalCodeAnalysisStage.cs

```diff
@@ -182,9 +222,12 @@ private static HashSet<IMethodDeclaration> FindHotRootMethods([NotNull]ICSharpFi
 
                         break;
                     case IMethodDeclaration methodDeclaration:
-                        // check that method is hot and add it to container                        
```

Comment:
It looks like the code above skips types nested in classlike declarations not inherited from `MonoBehaviour` which doesn't seem intended
e.g.
```cs
class C 
{
  class NotProcessed : MonoBehaviour { ... }
}
```

Reply:
It is not possible for Unity. MonoScript should be top-level class with same name as file name.